pkg/validation/options_test.go

```diff
@@ -140,6 +140,14 @@ func TestRedirectURL(t *testing.T) {
 	assert.Equal(t, expected, o.GetRedirectURL())
 }
 
+func TestUpstreamRequestTimeout(t *testing.T) {
+	o := testOptions()
+	o.UpstreamRequestTimeout = 40 * time.Second
+	assert.Equal(t, nil, Validate(o))
+	expected := 40 * time.Second
+	assert.Equal(t, expected, o.UpstreamRequestTimeout)
+}
+
```

Comment:
Are there any validations we should be performing on this value, for example, should we be checking that it is greater than zero.

Is there a way we can test that this is being set correctly on the http.Transport? Eg a test that creates a NewReverseProxy and checks that the right value is set

Reply:
I think more tests would be great. However I fear that implementing those might be not be possible in terms of time available and my current go knowledge.

It would be great if somebody could help with that. Maybe point to an example test that I could try to copy and modify?

Reply:
I will see if I can come up with some suggestions of how this might be tested over the weekend